src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3466,7 +3466,6 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
             {
                 switch (member)
                 {
-                    case FieldSymbol:
```

Comment:
> case FieldSymbol: [](start = 19, length = 18)

It looks like for fields the only thing we are going to compare is the Name. I think we should consider simply using a separate dictionary for that and avoiding introduction of SignatureOnlyFieldSymbol with the only purpose to lookup a field by name.